lib/bolt/pal.rb

```diff
@@ -490,16 +552,28 @@ def list_modules
       end
     end
 
-    def generate_types
+    def generate_types(cache: false)
       require 'puppet/face/generate'
       in_bolt_compiler do
         generator = Puppet::Generate::Type
         inputs = generator.find_inputs(:pcore)
         FileUtils.mkdir_p(@resource_types)
+        cache_plan_info if @project && cache
         generator.generate(inputs, @resource_types, true)
       end
     end
 
+    def cache_plan_info
+      # plan_name is an array here
+      plans_info = list_plans(filter_content: false).map do |plan_name|
+        data = get_plan_info(plan_name[0], with_mtime: true)
+        { plan_name[0] => data }
```

Comment:
```suggestion
      plans_info = list_plans(filter_content: false).map do |plan_name, |
        data = get_plan_info(plan_name, with_mtime: true)
        { plan_name => data }
```

I think this makes it slightly less confusing since `plan_name` won't be an array. Take it or leave it 🤷